### ISSUE TITLE
server/singleprocess: Update error messaging for polling

### ISSUE THREAD

**Issue by krantzinator** [hashicorp/waypoint#2202]
**Describe the bug**
The error message here could be updated to be more specific when the poll job fails because there are no active deployments or releases.
This should rarely happen, but for those edge cases, it shouldn't return an error.

**Steps to Reproduce**
* `waypoint init` a project
* Open the UI, and go to the project settings
* Fill out the git repo connection settings, and check "Enable builds on changes"
* The project won't deploy
* You'll eventually see the below in your server logs:
```
2021-09-02T18:52:13.972Z [TRACE] waypoint.server.singleprocess.poll_queuer.application_statusreport: returning peek for apps: project=go
2021-09-02T18:52:13.972Z [TRACE] waypoint.server.singleprocess.poll_queuer.application_statusreport: next poll time: time="2021-09-02 18:52:13.967891267 +0000 UTC m=+76.932768908"
2021-09-02T18:52:13.972Z [TRACE] waypoint.server.singleprocess.poll_queuer.application_statusreport: waiting on watchset and contexts
2021-09-02T18:52:13.972Z [TRACE] waypoint.server.singleprocess.poll_queuer.application_statusreport: queueing poll jobs
2021-09-02T18:52:13.972Z [TRACE] waypoint.server.singleprocess.poll_queuer.application_statusreport.go.go: looking at latest deployment and release to generate status report on
2021-09-02T18:52:13.972Z [WARN]  waypoint.server.singleprocess.poll_queuer.application_statusreport: error building a poll job request. This should not happen repeatedly. If you see this in your log repeatedly, report a bug.: err="rpc error: code = NotFound desc = No application named "go" is available!"
```

I used https://github.com/briancain/go-gitops on Waypoint 0.5.1

**Expected behavior**
Waypoint should not return an error on polling if there are no deploys/releases to poll.

**Waypoint Platform Versions**
Additional version and platform information to help triage the issue if
applicable:

$ waypoint version
CLI: v0.5.1 (76d7e17f)
Server: v0.5.1

**Additional context**
This is low priority.

**Comment by briancain** [comment]
We're about to fix this one, but this issue has a couple of things that need to be fixed:

- A recent update to the app status poll handler last release always returned an error if no deployments or releases were found. In this case, we already have some handling a few lines later that would skip building the job rather than returning an error.
- The AppStatusPeek function shouldn't be returning an app to poll on if there's no deployments that were made, even if the poll time is zero. Fixing this means we won't be building a poll job on an app with no deployments.
